certification/internal/policy/operator/scc_check.go

```diff
@@ -0,0 +1,121 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/bundle"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+)
+
+// SecurityContextConstraintsCheck evaluates the csv and logs a message if a non default security context constraint is
+// needed by the operator
+
+type SecurityContextConstraintsCheck struct {
+	customSCCSpecified  bool
+	builtInSCCSpecified bool
+}
+
+func getDefaultSCCs() []string {
+	return []string{"restricted", "privileged", "nonroot", "node-exporter", "hostnetwork", "hostmount-anyuid", "hostaccess", "anyuid"}
+}
+
+func NewSecurityContextConstraintsCheck() *SecurityContextConstraintsCheck {
+	return &SecurityContextConstraintsCheck{
+		false,
+		false,
+	}
+}
+
+func (p *SecurityContextConstraintsCheck) Validate(ctx context.Context, bundleRef certification.ImageReference) (bool, error) {
+	requestedSCCList, err := p.dataToValidate(ctx, bundleRef.ImageFSPath)
+	if err != nil {
+		return false, err
+	}
+
+	return p.validate(ctx, requestedSCCList)
+}
+
+func (p *SecurityContextConstraintsCheck) dataToValidate(ctx context.Context, imagePath string) ([]string, error) {
+	csvFilepath, err := bundle.GetCsvFilePathFromBundle(imagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	csvFileReader, err := os.Open(csvFilepath)
+	if err != nil {
+		return nil, err
+	}
+
+	requestedSccList, err := bundle.GetSecurityContextConstraints(ctx, csvFileReader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract security context constraints from ClusterServiceVersion: %w", err)
+	}
+
+	return requestedSccList, nil
+}
+
+func (p *SecurityContextConstraintsCheck) validate(ctx context.Context, requestedSccList []string) (bool, error) {
+	defaultSccList := getDefaultSCCs()
+
+	if requestedSccList == nil || len(requestedSccList) == 0 {
+		log.Infof("No security context constraint was requested so using the default %s scc", defaultSccList[0])
+		return true, nil
+	}
+
+	if len(requestedSccList) != 1 {
+		return false, fmt.Errorf("only one scc should be requested at a time")
+	}
+
+	for _, defaultScc := range defaultSccList {
+		if requestedSccList[0] == defaultScc {
+			log.Infof("A built in default security context constraint %s was requested", defaultScc)
+			p.builtInSCCSpecified = true
+			return true, nil
+		}
+	}
+
+	log.Infof("A custom scc was specified: %s", requestedSccList[0])
```

Comment:
This is a log-only check, right? So if we are not planning on including it in the results.json, we need the log line to contain all of the useful information for the user.

I see you setting the p.customSCCSpecified bool and using that to drive Metadata/Help, but I don't think the user will ever see those if I'm understanding how this check is supposed to play out.

Reply:
Yes currently these are not shown anywhere.. but was hoping we could have a level called "informational" or something along those lines that would show the appropriate help text at some point ? don't think logs are sufficient to indicate things to the end user.. I imagine end users should be indicated to with actual output ?

Reply:
> but was hoping we could have a level called "informational" or something along those lines that would show the appropriate help text at some point

+1 That's fine, but until that happens, we need to log all of the context. I'd also leave a `// TODO` comment in with a note indicating what you mentioned to be resolved at a later date.

Reply:
added and updated log messaged 👍